src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -89,4 +89,43 @@ public static Entity createGameEntity(String gameName, DatastoreService datastor
  
     return entity;
   }
+
+  /**
+  * add user to user list in game entity, and creates a score entity for the given user and game
+  */
+  public static boolean addUserToGame(String userId, String userName, Entity gameEntity, DatastoreService datastore) {
+ 
+    if(userId == "" || userName == ""){
+      return false;
+    }
+
+    if(gameEntity == null || datastore == null){
+      log.severe("null entity or datastore"); 
+      return false;
+    }
+ 
+    ArrayList<String> userIds = (ArrayList<String>) gameEntity.getProperty("userIds");
+    if(userIds == null){
+      log.severe("empty user id list");
+      return false;
+    }
+ 
+    // create score entity for the user for this game (starts as 0)
+    int score = 0;
+    Entity scoreEntity = new Entity("Score");
+    scoreEntity.setProperty("userId", userId);
+    scoreEntity.setProperty("userName", userName);
```

Comment:
Are you sure you need to store the userName in the Score entity? I recommend holding off on this until you decide you need it. Which I suspect you won't, because you can use the userId to look up the userName.

(My point is that if you avoid storing the same information in multiple places, the system will be simpler and you'll run into fewer problems later.)

Reply:
That's a good point, I took it out